docs/guide/apm-data-model.asciidoc

```diff
@@ -144,4 +157,100 @@ For a full list of tracked metrics, see the relevant agent documentation:
 * {apm-java-ref-v}/metrics.html[Java]
 * {apm-node-ref-v}/metrics.html[Node.js]
 * {apm-py-ref-v}/metrics.html[Python]
-* {apm-ruby-ref-v}/metrics.html[Ruby]
\ No newline at end of file
+* {apm-ruby-ref-v}/metrics.html[Ruby]
+
+// This heading is linked to from the APM UI section in Kibana
+[[metadata]]
+=== Metadata
+
+Metadata can enrich your events and make application performance monitoring even more useful.
+Let's explore the different types of metadata that Elastic APM offers.
+
+==== Quick Reference table
+
+[options="header", cols="1,1,1,2"]
+|=======================================================================
+|Metadata   |Indexed   |Applies to     |Agent API links
+
+|Labels
+|Yes
+v|<<transactions>>
+<<transaction-spans>>
+<<errors>>
+v|Go: {apm-go-ref}/api.html#context-set-tag[`SetTag`]
+Java: {apm-java-ref}/public-api.html#api-transaction-add-tag[`addLabel`]
+Node.js: {apm-node-ref}/agent-api.html#apm-set-tag[`setTag`] \| {apm-node-ref}/agent-api.html#apm-add-tags[`addTags`]
+Python: {apm-py-ref}/api.html#api-tag[`tag`]
+Ruby: {apm-ruby-ref}/api.html#api-agent-set-tag[`set_tag`]
+Rum: {apm-rum-ref}/api.html#apm-add-tags[`addTags`]
+
+|Custom
+|No
+v|<<transactions>>
+<<errors>>
+v|Go: {apm-go-ref}/api.html#slug-here[`???`]
+Java: {apm-java-ref}/public-api.html#slug-here[`???`]
```

Comment:
@felixbarny - The java agent has no custom context?

Reply:
No, not yet, but as we resolved  elastic/apm#7, we can add it now: https://github.com/elastic/apm-agent-java/issues/604